wordpress-mobile/WordPress-iOS
# Fix broken Core Data model 75 with featuredOnPosts

This PR fixes a problem with migrating from model 74 (WPiOS 9.9's last model) to model 76 (released as 10.0.0.20180511 & 10.0.1.2). Apparently Core Data says it'll support migrating a relationship destination entity change but in reality it causes an exception to happen in SQLite, aborting the migration. Anyone upgrading to 10.0.0.0 would get forcibly logged out.

This PR merges the changes from 76 into 75 and deletes model 76. This ensures production 9.9 to 10.0 is seamless. 

**Caveat:** Deleting model 76 does not prune the directory because of Git being stupid. After you check out this PR (and anyone else trying to build after this is merged into develop) must do:

`rmdir ./WordPress/Classes/WordPress.xcdatamodeld/WordPress\ 76.xcdatamodel`

To test:
1. Clean, Build, Run unit tests.
2. Check out `release/10.0` (or tag 9.9). `pod install`, clean, build, run, log in. Kill the app.
3. Check out this PR again. `pod install`, clean, build, run. Verify you're still logged in.

Sorry for all the bad advice @frosty!
